# t-test

## one sample t-test

### Case 1: average Battery Lifetime

- H0: battery lifetime is 1000 hours
- H1: battery lifetime is not 1000 hours

### Data

In [15]:
battery <- c(980, 1008, 968, 930, 1012, 1002, 996, 1017)

#### Nomality test

- H0: the population is normally distributed.  
- H1: the population is NOT normally distributed.

In [16]:
shapiro.test(battery)


	Shapiro-Wilk normality test

data:  battery
W = 0.87452, p-value = 0.1668


    : p-value = 0.9469  > 0.05, H0 is accepted. 
    The data is normally distributed.

### t-test

In [17]:
t.test(battery, mu=1000, alternative = "two.sided")

# alternative = "two.sided", "less", "greater"


	One Sample t-test

data:  battery
t = -1.06, df = 7, p-value = 0.3244
alternative hypothesis: true mean is not equal to 1000
95 percent confidence interval:
  964.8642 1013.3858
sample estimates:
mean of x 
  989.125 


    : As p-value 0.3244 > 0.05, alternatively hypothesis(H1) 'battery lifetime is NOT 1000' is accepted.

### Case 2: average math exam

- H0: average of math score is 55
- H1: average of math score greater than 55

### Data

In [24]:
math <- c(58, 49, 39, 99, 32, 88, 62, 30, 55, 65, 44, 55, 57, 53, 88, 42, 39)
math
summary(math)

[1] 58 49 39 99 32 88 62 30 55 65 44 55 57 53 88 42 39

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  30.00   42.00   55.00   56.18   62.00   99.00 

In [25]:
shapiro.test(math)


	Shapiro-Wilk normality test

data:  math
W = 0.91143, p-value = 0.1058


     : p-value = 0.9469  > 0.05, H0 is accepted. The data is normally distributed.

### t.test

- H0: mean is NOT greater than 55
- H1: mean is greater than 55

In [20]:
t.test(math, mu=55, alternative = "greater")


	One Sample t-test

data:  math
t = 0.24546, df = 16, p-value = 0.4046
alternative hypothesis: true mean is greater than 55
95 percent confidence interval:
 47.80855      Inf
sample estimates:
mean of x 
 56.17647 


    : Since p-value 0.4046 is 0.05, H0 is accepted. This is because the average is distored by outliers (e.g. score 99).

## two sample t-test

### case 3: medicinal efficacy test 

### Data

In [ ]:
H0: The medicine does not improve health index.
H1: The medicine improves health index.

In [27]:
pre  <- c(13.2, 8.2, 10.9, 14.3, 10.7, 6.6, 9.5, 10.8, 8.8, 13.3)
post <- c(14,   8.8, 11.2, 14.2, 11.8, 6.4, 9.8, 11.3, 9.3, 13.6)

In [28]:
summary(pre)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  6.600   8.975  10.750  10.630  12.625  14.300 

In [29]:
summary(post)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  6.400   9.425  11.250  11.040  13.150  14.200 

### t-test

In [30]:
t.test(pre, post)


	Welch Two Sample t-test

data:  pre and post
t = -0.36891, df = 17.987, p-value = 0.7165
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -2.745046  1.925046
sample estimates:
mean of x mean of y 
    10.63     11.04 


    : While the average seems to be improved, the null hypothesis(The medicine does not improve health index) is acceptable because of p-value 0.7165 > 0.05.